operator/watchers/k8s_service_sync.go

```diff
@@ -186,15 +185,17 @@ func StartSynchronizingServices(ctx context.Context, clientset k8sClient.Clients
 		fallthrough
 	default:
 		endpointController = endpointsInit(clientset.Slim(), swgEps, serviceOptsModifier)
-		go endpointController.Run(wait.NeverStop)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			endpointController.Run(ctx.Done())
+		}()
 	}
 
+	wg.Add(1)
 	go func() {
-		<-k8sSvcCacheSynced
-		cache.WaitForCacheSync(wait.NeverStop, endpointController.HasSynced)
-	}()
+		defer wg.Done()
```

Comment:
What's going on in this hunk? It seems like the previous version waited for the cache to be synced, while the new one doesn't delay at all. As far as I could follow, the rest of the WaitGroup handling was set up to ensure that all cleanup completes before the next test runs, which doesn't seem related to removing the cache sync logic here.

Reply:
This was some left-over from earlier and it wasn't doing anything after waiting for sync, hence I removed it.